pkg/controller/servingcert/controller/secret_creating_controller.go

```diff
@@ -313,11 +322,17 @@ func getServingCert(dnsSuffix string, ca *crypto.CA, service *corev1.Service) (*
 	if err != nil {
 		return nil, err
 	}
+	// Including the intermediate cert will ensure that clients with a
+	// stale ca bundle (containing the previous CA but not the current
+	// one) will be able to trust the serving cert.
+	if intermedicateCACert != nil {
+		servingCert.Certs = append(servingCert.Certs, intermedicateCACert)
+	}
 	return servingCert, nil
 }
 
-func toRequiredSecret(dnsSuffix string, ca *crypto.CA, service *corev1.Service, secretCopy *corev1.Secret) error {
-	servingCert, err := getServingCert(dnsSuffix, ca, service)
+func toRequiredSecret(dnsSuffix string, ca *crypto.CA, intermediateCACert *x509.Certificate, service *corev1.Service, secretCopy *corev1.Secret) error {
+	servingCert, err := getServingCert(dnsSuffix, ca, intermediateCACert, service)
```

Comment:
The original PR was not adding the intermediate to the serving cert. I think commonly the server just sends its whole chain. I wonder whether this step would actually make a server which would otherwise send just its own cert send both or not.

Reply:
The server only sends the certs that are supplied to it. Typically that won't include an entire trust chain, but only the certs required for the client to verify from what the client already knows (i.e. a ca bundle). The goal here is ensuring that a server using a serving cert will send clients `#4` and `Y` so that unrefreshed clients will be able to verify `Y` against an old bundle that contains `#1` but is missing `#2` (because `Y` -> `#4` -> `#1`).

I talked to Matt yesterday and confirmed that the original PR was in error in not adding the intermediate (`#4`) to new serving certs. The design doc even explicitly says it is required - `Distribute #4 with the new service-certs (appended to PEM)`.

I've also confirmed this in the unit testing - try removing [the addition of the intermediate cert](https://github.com/openshift/service-ca-operator/pull/73/files#diff-c482e5ba51661f6d2b95bcc56e0cd468R59) and the `Server rotated` test scenario will fail as expected.